### PR TITLE
adobe

### PR DESCRIPTION
**Pull request by GurgenBaghdasaryan**

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import React from "react";
+import "./App.css";
+import Login from "./components/Login/index";
+import User from "./components/User/index";
+import { Route, Switch } from "react-router-dom";
```
Comment (arayik-99) [inline src/App.js]: When importing, please firstly import the packages (react, react-route-dom etc.), then the ones.

File: src/App.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import React from "react";
+import "./App.css";
```
Comment (edotnet) [inline src/App.js]: move css import to end

File: src/App.js (hunk 1 above)
Comment (edotnet) [inline src/App.js]: move package import to top

File: src/App.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+import Login from "./components/Login/index";
+import User from "./components/User/index";
+import { Route, Switch } from "react-router-dom";
+
+const App = () => {
+  return (
+    <main className='main'>
+      <Switch>
+        <Route path="/" component={Login} exact />
```
Comment (edotnet) [inline src/App.js]: fix paths

File: src/UserContext/index.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
+import React, { createContext, useState } from "react";
+
+export const UserContext = createContext();
+
+const UserProvider = ({ children }) => {
+  const [loginField, setLoginField] = useState("");
+  const [passwordField, setPasswordField] = useState("");
+  const values = {
+    login: "gug12321",
+    password: "gug12321",
```
Comment (arayik-99) [inline src/UserContext/index.js]: You don't need these hardcoded login/password anymore

File: src/components/Login/index.jsx [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+const Login = () => {
+  const { loginField, passwordField, login, password } = useContext(
+    UserContext
+  );
+
+  const history = useHistory();
+  const logIn = () => {
+    if (!loginField.trim() || !passwordField.trim()) return;
+    if (login !== loginField && passwordField !== password) return;
```
Comment (arayik-99) [inline src/components/Login/index.jsx]: This is also not needed

Comment (edotnet) [inline src/components/Login/index.jsx]: remove this checking since its done in backend

File: src/components/Login/index.jsx [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+  
+    axios
+      .post("https://volatile-admin-api.herokuapp.com/auth/login", {
+        username: login,
+        password: password,
+      })
+      .then((result) => {
+        localStorage.setItem('access_token', result.data.access_token);
```
Comment (edotnet) [inline src/components/Login/index.jsx]: save token in context

File: src/components/LoginField/index.jsx [hunk 7]
```diff
@@ -0,0 +1,39 @@
+import React, { useContext, useCallback } from "react";
+import { UserContext } from "../../UserContext";
+import { StyledInput, StyledParagraph } from "./styles";
+import Button from "../Button/index";
+
+const LoginField = ({ loginEvent }) => {
+  const { setPasswordField, setLoginField } = useContext(UserContext);
+
+  // const check = () =>{
```
Comment (edotnet) [inline src/components/LoginField/index.jsx]: remove comments

File: src/components/LoginField/index.jsx [hunk 8]
```diff
@@ -0,0 +1,39 @@
+import React, { useContext, useCallback } from "react";
+import { UserContext } from "../../UserContext";
+import { StyledInput, StyledParagraph } from "./styles";
+import Button from "../Button/index";
+
+const LoginField = ({ loginEvent }) => {
+  const { setPasswordField, setLoginField } = useContext(UserContext);
```
Comment (edotnet) [inline src/components/LoginField/index.jsx]: use state for local fields

File: src/components/LoginField/styles.js [hunk 9]
```diff
@@ -0,0 +1,14 @@
+import styled from "styled-components";
+
+export const StyledInput = styled.input`
+  width: 408px;
+  height: 48px;
+  margin-bottom: 24px;
+  border: 1px solid #00000026;
+border-radius: 2px;
```
Comment (edotnet) [inline src/components/LoginField/styles.js]: format

File: src/components/LoginField/index.jsx [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+
+  // const check = () =>{
+  //   if(! && !){
+  //     return <p>Incorrect username or password</p>;
+  //   }
+  // }
+
```
Comment (arayik-99) [inline src/components/LoginField/index.jsx]: Please, don't commit commented code (Or commit with statement above it)

File: src/components/User/index.jsx [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+import { useHistory } from "react-router-dom";
+ import {UserContext} from "../../UserContext";
+
+const User = () => {
+  const {login , password } = useContext(UserContext);
+
+  const history = useHistory();
+
+  const logIn = () => {    
+    history.push('/')
```
Comment (edotnet) [inline src/components/User/index.jsx]: here you should remove token from context which should cause to redirect onto login page

File: src/index.js [hunk 12]
```diff
@@ -0,0 +1,21 @@
+import React from "react";
+import { BrowserRouter } from "react-router-dom";
+import ReactDOM from "react-dom";
+import "./index.css";
```
Comment (edotnet) [inline src/index.js]: move to end

File: src/components/LoginField/index.jsx [hunk 13]
```diff
@@ -0,0 +1,39 @@
...
+      />
+      <StyledParagraph>Password</StyledParagraph>
+       {/* <p>{check()}</p>  */}
+      <StyledInput
+        onChange={(e) => setPasswordField(e.target.value)}
+        type="password"
+        style={{ fontSize: "20px", textIndent: "5px" }}
+      />
+      <br />
+
```
Comment (arayik-99) [inline src/components/LoginField/index.jsx:23]: Unnesecarry line.

File: src/UserContext/index.js [hunk 14]
```diff
@@ -0,0 +1,20 @@
+import React, { createContext, useState } from "react";
+
+export const UserContext = createContext();
+
+const UserProvider = ({ children }) => {
+  const [loginField, setLoginField] = useState("");
+  const [passwordField, setPasswordField] = useState("");
+  const values = {
+    login: "gug12321",
```
Comment (edotnet) [inline src/UserContext/index.js]: remove all fields from here and add token and isAuthenticated fields

**Review by edotnet — CHANGES_REQUESTED** [review]

File: src/components/User/index.jsx [hunk 15]
```diff
@@ -0,0 +1,25 @@
+import React,{useContext} from "react";
+import { StyledButton } from "./styles";
+import { useHistory } from "react-router-dom";
+ import {UserContext} from "../../UserContext";
```
Comment (arayik-99) [inline src/components/User/index.jsx]: Unnesecarry whitespace.

File: src/components/User/index.jsx [hunk 16]
```diff
@@ -0,0 +1,25 @@
...
+
+  const logIn = () => {    
+    history.push('/')
+  };
+
+  return (
+    <>
+      <p>{login}</p>
+      <p>{password}</p>
+
```
Comment (arayik-99) [inline src/components/User/index.jsx]: Unnesecarry line.

File: src/UserContext/index.js [hunk 17]
```diff
@@ -3,17 +3,14 @@ import React, { createContext, useState } from "react";
 export const UserContext = createContext();
 
 const UserProvider = ({ children }) => {
-  const [loginField, setLoginField] = useState("");
-  const [passwordField, setPasswordField] = useState("");
+  const getToken = localStorage.getItem('access_token');
+  
+  const [token, setToken] = useState(getToken);
```
Comment (arayik-99) [inline src/UserContext/index.js:6]: Please, when using useState, put it in the top of component.

File: src/components/LoginField/index.jsx [hunk 18]
```diff
@@ -1,18 +1,10 @@
...
+const LoginField = ({ loginEvent ,values}) => {
 
-  // const check = () =>{
-  //   if(! && !){
-  //     return <p>Incorrect username or password</p>;
-  //   }
-  // }
-
- 
+  const {setLoginField,setPasswordField} = values;
```
Comment (arayik-99) [inline src/components/LoginField/index.jsx]: Unnesecary line, delete the above line, add one line below this

File: src/components/User/index.jsx [hunk 19]
```diff
@@ -1,23 +1,25 @@
-import React,{useContext} from "react";
+import React, { useContext } from "react";
 import { StyledButton } from "./styles";
 import { useHistory } from "react-router-dom";
- import {UserContext} from "../../UserContext";
+import {UserContext} from '../../UserContext/index'
+
```
Comment (arayik-99) [inline src/components/User/index.jsx]: Unnsecarry line

File: src/components/User/index.jsx [hunk 20]
```diff
@@ -1,23 +1,25 @@
...
 
 const User = () => {
-  const {login , password } = useContext(UserContext);
+  const {setToken} = useContext(UserContext)
```
Comment (arayik-99) [inline src/components/User/index.jsx]: Put whitespaces before after setToken

**Pull request merged**